jersey/src/main/java/com/netflix/conductor/server/resources/ApplicationExceptionMapper.java

```diff
@@ -47,38 +50,58 @@
 public class ApplicationExceptionMapper implements ExceptionMapper<ApplicationException> {
 
 	private static Logger logger = LoggerFactory.getLogger(ApplicationExceptionMapper.class);
-	
-	private static List<Variant> supportedMediaTypes = Variant.mediaTypes(MediaType.APPLICATION_JSON_TYPE, MediaType.TEXT_HTML_TYPE, MediaType.TEXT_PLAIN_TYPE).add().build();
-	
-	@Context 
-    private HttpContext context;
-	
+
+	private static List<Variant> supportedMediaTypes = Variant.mediaTypes(MediaType.APPLICATION_JSON_TYPE,
+			MediaType.TEXT_HTML_TYPE, MediaType.TEXT_PLAIN_TYPE).add().build();
+
+	@Context
+	private HttpContext context;
+
+	@Context
+	private UriInfo uriInfo;
+
+	@Context
+	private javax.inject.Provider<Request> request;
+
 	private String host;
-	
+
 	@Inject
 	public ApplicationExceptionMapper(Configuration config) {
 		this.host = config.getServerId();
 	}
-	
+
 	@Override
 	public Response toResponse(ApplicationException e) {
-		logger.error(e.getMessage(), e);
+		logException(e);
+
+		Response.ResponseBuilder responseBuilder = Response.status(e.getHttpStatusCode());
+
 		if(e.getHttpStatusCode() == 500) {
 			Monitors.error("error", "error");
 		}
-		MediaType mediaType = context.getRequest().selectVariant(supportedMediaTypes).getMediaType();
-		if(mediaType == null){
-			mediaType = MediaType.APPLICATION_JSON_TYPE;
-		}
-		
+
 		Map<String, Object> entityMap = e.toMap();
 		entityMap.put("instance", host);
-		Object entity = entityMap;
-		
-		if (mediaType == null || MediaType.APPLICATION_JSON_TYPE != mediaType) {
-			entity = e.toMap().toString();
-		}
-		return Response.status(e.getHttpStatusCode()).entity(entity).type(mediaType).build();
+
+		responseBuilder.type(MediaType.APPLICATION_JSON_TYPE);
+		responseBuilder.entity(entityMap);
+
+		return responseBuilder.build();
+	}
+
+	@VisibleForTesting
+	UriInfo getUriInfo() {
+		return uriInfo;
+	}
+
+	@VisibleForTesting
+	Request getRequest() {
+		return request.get();
+	}
+
+	private void logException(ApplicationException exception) {
+		logger.error(String.format("%s message= '%s' url= '%s'", exception.getClass().getSimpleName(),
```

Comment:
This would not log the exception stack trace. You might want to do:
```java
logger.error(String.format("Error in %s url= '%s'", exception.getClass().getSimpleName(), getUriInfo().getPath()), exception);
```